sandcastle/api.py

```diff
@@ -502,26 +502,45 @@ def _do_exec(
                 continue
         raise SandcastleException("Unable to connect to the kubernetes API server.")
 
-    def _prepare_mdir_exec(
-        self, command: List[str], target_dir: Optional[Path] = None
+    def _prepare_exec(
+        self,
+        command: List[str],
+        target_dir: Optional[Path] = None,
+        env: Optional[Dict] = None,
+        cwd: Union[str, Path] = None,
     ) -> Tuple[Path, Path]:
         """
-        wrap a command to exec in the sandbox in a script
+        wrap a command for exec into a script file and place it to sandbox
 
         :param command: command to wrap
         :param target_dir: a dir in the sandbox where the script is supposed to be copied
+        :param env: env vars to set in the script.sh so they are available to the "command"
+        :param cwd: run the command in this directory, defaults to a mapped dir
+               or a temporary directory if mapped_dir is not set
         :return: (path to the sync'd dir within sandbox, path to the script within sandbox)
         """
         cmd_str = " ".join(f"{shlex.quote(c)}" for c in command)
 
-        root_target_dir = Path(target_dir or self._do_exec(["mktemp"]).strip())
+        root_target_dir = Path(target_dir or self._do_exec(["mktemp", "-d"]).strip())
         # this is where the content of mapped_dir will be
         unique_dir = root_target_dir / self.pod_name
         script_name = "script.sh"
 
-        # git checkout - oc cp does not preserve the file mode and makes the repo dirty
         # set -e - fail the script when a command fails
-        script_template = f"#!/bin/bash\nset -e\ncd {unique_dir}\nexec {cmd_str}\n"
+        script_template = (
+            "#!/bin/bash\n"
+            "set -e\n"
+            "source /home/sandcastle/setup_env_in_openshift.sh\n"
+            f"mkdir -p {unique_dir}\n"
+        )
+        cwd = cwd or ""
+        script_template += f"cd {unique_dir}/{cwd}\n"
```

Comment:
Reading these lines together makes me wondering: if the script is supposed to create `unique_dir`, how is `uniquedir/cwd` created?

Reply:
the expectation is that cwd is going to be a subdir in a git repo coming from packit-service (though packit is not using cwd argument for user commands anywhere - all of them run from the root of the repo)

as I wrote in the commit message, this feature is not very useful and is implemented only to match the [CommandHandler](https://github.com/packit/packit/blob/main/packit/command_handler.py#L118) interface in packit

I can reorder the lines and do `f"mkdir -p {unique_dir}/{cwd}\n"`, thanks for pointing this out.